### ISSUE TITLE
Filtering logic for unsupported: ovm crashes hardhat plugin with `No input sources specified`

### ISSUE THREAD

**Issue by cgewecke** [ethereum-optimism/optimism#674]
**Describe the bug**
Hardhat sometimes automatically partitions the compilation job set into several sections.

If you are:
+ only whitelisting a subset of contracts for ovm 
+ a compilation job hardhat composed contains no whitelisted contracts

...HH will throw a `No input sources...` error 
```
Compiling 148 files with 0.6.12
OVM Compiler Error (insert "// @unsupported: ovm" if you don't want this file to be compiled for the OVM):
 No input sources specified.
Error HH600: Compilation failed
```

The problem is here, where filtering can result in an empty input passed to `TASK_COMPILE_SOLIDITY_RUN_SOLCJS`.

https://github.com/ethereum-optimism/optimism/blob/4c2fbdeac31908e240774723ba69a6a381481d47/packages/hardhat-ovm/src/index.ts#L153-L168

Was able to patch this locally by returning an empty object if there are no sources.

```diff
- // Build both inputs separately.
+ if (Object.keys(ovmInput.sources).length === 0) return {};
+ // Build both inputs separately.
```

**To Reproduce**
Apologies, I don't have simple reproduction steps. This affects larger projects which hardhat subdivides compilation into several jobs for.  

**Expected behavior**
Compilation succeeds. 

**System Specs:**
 - OS: osx
 - Package Version (or commit hash): `eth-optimism/hardhat-ovm 0.0.3`

**Comment by tynes** [comment]
cc @smartcontracts

**Comment by snario** [comment]
Fixed by #688
